### ISSUE TITLE
Deprecate [leadership] indefinitely

### ISSUE THREAD

**Issue by soliton-** [wesnoth/wesnoth#8215]
### Describe the desired feature

The leadership concept existed in wesnoth since forever and is easy to understand intuitively.

The `[leadership]` ability tag does not reflect that concept by default. Instead it requires `affect_self=no` and `[affect_adjacent]` which makes any ability leadership-like... This is confusing.

It would be clearer to implement leadership abilities using `[damage]` and deprecate `[leadership]` indefinitely. The goal is in the spirit of #8030 to make abilities easier to understand. IMO a tag name like leadership that implies a well-known concept but does not actually embody it just makes everything more confusing.

If it is not possible to implement leadership with `[damage]` that seems like either behavior differences that should not exist and need to be examined closely or missing features that it would be worthwhile to implement (probably not just for `[damage]`).

(Of course I'm not suggesting this for 1.18 but for 1.19 or later.)

**Comment by gfgtdf** [comment]
Leadersship has some small differences to damage

1) it shows the leadership animations
2) the damage calculation is probably a little different (the way it stack with other damage abilities)
3) I _think_ its in theory also active outside of battles, not sure whether this actually changes something.

None of these are unsolvable but something to keep in mind when doing this.

**Comment by ProditorMagnus** [comment]
3. is same for [damage] and [leadership]

**Comment by CelticMinstrel** [comment]
While we're on the topic of redundant ability tags, I also want to draw attention to `[regenerate]`, which does the same thing as `[heals]`.

**Comment by newfrenchy83** [comment]
@doofus-01 [teaching_anim] is the equivalent of [leading_anim] for weapons used as leadership like chance_to_hit and [damage] with aply_to=self, and the equivalent of resistance_anim when apply_to=opponent, it is therefore already possible to use damage leadership to replace [ leadership] with an animation. The main problem lies in the method of calculating each of them and the fact that [damage] is calculated before that of the time of day influence while the calculation by [leadership] is carried out afterwards. This is also why I prefer that [leadership] be preserved.

**Comment by doofus-01** [comment]
>The main problem lies in the method of calculating each of them and the fact that [damage] is calculated before that of the time of day influence while the calculation by [leadership] is carried out afterwards. This is also why I prefer that [leadership] be preserved.

I agree, that does seem to support not removing [leadership].  Or else [damage] needs more customization options, which won't make things any simpler or clearer.  

The `[teaching_anim]` stuff doesn't strike me as intuitive, maybe because it doesn't have the directly named link to an ability or event (like `[leading_anim]`/`[leadership]`, `[movement_anim]`/`movement`/`moveto` vs `[teaching_anim]`/`[damage]`).  Removing `[leadership]` to end up with `[leading_anim]`/`[damage]` or `[teaching_anim]`/`[damage] for use with leadership concept` would not make things clearer.  It's nice that it's possible to use `[teaching_anim]` for a "leadership" ability, but this would not be an improvement.

**Comment by Pentarctagon** [comment]
I think part of the problem is just describing these abilities as "leadership-like" or "used as leadership", since what it really means is "any ability that uses [affect_adjacent]".

**Comment by soliton-** [comment]
Just to be clear [indefinite deprecation](https://wiki.wesnoth.org/CompatibilityStandards#Deprecation_levels_-_When_to_remove_deprecated_features) does not mean removing.

To me describing an ability as "leadership-like" means it works like leadership so there is a leading unit and another unit that receives the effect from the leader. That is pretty much what "an ability that uses [affect_adjacent]" means, right? To me that part is intuitive and good.

Now another quite likely understanding is that "leadership-like" means like the `[leadership]` tag which is unfortunately not true and exactly why I opened this issue.

---

> > The main problem lies in the method of calculating each of them and the fact that [damage] is calculated before that of the time of day influence while the calculation by [leadership] is carried out afterwards. This is also why I prefer that [leadership] be preserved.
> 
> I agree, that does seem to support not removing [leadership]. Or else [damage] needs more customization options, which won't make things any simpler or clearer.
 
I disagree, it absolutely will make it clearer if there are explicit attributes to govern damage calculation instead of dubious implicit differences between `[damage]` and `[leadership]` that rather sound like bugs and don't appear to be documented.

> The `[teaching_anim]` stuff doesn't strike me as intuitive, maybe because it doesn't have the directly named link to an ability or event (like `[leading_anim]`/`[leadership]`, `[movement_anim]`/`movement`/`moveto` vs `[teaching_anim]`/`[damage]`). Removing `[leadership]` to end up with `[leading_anim]`/`[damage]` or `[teaching_anim]`/`[damage] for use with leadership concept` would not make things clearer. It's nice that it's possible to use `[teaching_anim]` for a "leadership" ability, but this would not be an improvement.

I don't know why there is just `[teaching_anim]` for all abilities besides leadership and resistance but there could certainly be `[damage_anim]` and whatnot.

**Comment by newfrenchy83** [comment]
> I don't know why there is just `[teaching_anim]` for all abilities besides leadership and resistance but there could certainly be `[damage_anim]` and whatnot.

encode each weapon special animation with their proper tags is too redundant to do.

**Comment by CelticMinstrel** [comment]
I think it's better to have a `[teaching_anim]` with a key specifying the ability it applies to (including `resistance` too!) than one animation tag for every possible ability, because that makes it easier to add a new ability later. I'm not sure if that's how `[teaching_anim]` currently works.
